### PR TITLE
feat: add method for decoding sealed results

### PR DESCRIPTION
**Pull request by TheUnderScorer**

### CODE REVIEW

File: src/sealedResults.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    throw new Error('Invalid sealed data header');
+  }
+
+  for (const decryptionKey of decryptionKeys) {
+    switch (decryptionKey.algorithm) {
+      case DecryptionAlgorithm.Aes256Gcm:
+        try {
+          return await unsealAes256Gcm(sealedData, decryptionKey.key);
+        } catch {
+          continue;
```
Comment (ilfa) [inline src/sealedResults.ts:43]: We don't provide good error reporting here. It would be good to know what the problem was if decryption failed for all keys. 
Let's collect exceptions to an array and instead of just `Unable to decrypt sealed data` provide a better error. Like the first key was not the right length, the others were just wrong.

File: readme.md [hunk 2]
```diff
@@ -145,6 +145,11 @@ client
   });
 ```
 
+## Sealed results
+
+This SDK provides utility methods for decoding sealed results.
+To learn more, refer to example located in [example/sealedResults.js](./example/sealedResults.js).
```
Comment (ilfa) [inline readme.md:151]: Same with go. Let's add an example of usage to the readme.
And we will need to add method description to the [API reference](https://github.com/fingerprintjs/fingerprintjs-pro-server-api-node-sdk?tab=readme-ov-file#api-reference) section

File: tests/unit-tests/sealedResults.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        },
+        {
+          key: validKey,
+          algorithm: DecryptionAlgorithm.Aes256Gcm,
+        },
+      ])
+    ).rejects.toThrowError('Invalid sealed data header');
+  });
+
+  it('throws error if all decryption keys are invalid', async () => {
```
Comment (ilfa) [inline tests/unit-tests/sealedResults.ts:126]: We need to add key with a wrong size as well.

**Review by ilfa — CHANGES_REQUESTED** [review]
Please improve code coverage for sealed results. It should be easy to make it 100% by adding tests for all negative scenarios.

File: example/sealedResults.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+const { unsealEventsResponse } = require('@fingerprintjs/fingerprintjs-pro-server-api');
+
+async function main() {
+  const sealedData = process.env.BASE64_SEALED_RESULT;
```
Comment (makma) [inline example/sealedResults.js:4]: Why did u decide to put this into the env var?

File: src/sealedResults.ts (hunk 1 above)
Comment (TheUnderScorer) [reply src/sealedResults.ts:57]: Done!

File: readme.md (hunk 2 above)
Comment (TheUnderScorer) [reply readme.md:151]: Done!

File: example/sealedResults.js (hunk 4 above)
Comment (TheUnderScorer) [reply example/sealedResults.js:4]: What other approach would you suggest? It's meant for customers to test it, similar to example in `index.js`

Comment (makma) [reply example/sealedResults.js:4]: Oh, I see, it's just a node function. I think this is fine, thanks

File: readme.md [hunk 5]
```diff
@@ -409,6 +414,49 @@ For more information, see the [Server API documentation](https://dev.fingerprint
 }
 ```
 
+## Sealed results API Reference
+
+##### `unsealEventsResponse(sealedData: Buffer, decryptionKeys: DecryptionKey[]): Promise<EventResponse>`
```
Comment (ilfa) [inline readme.md:419]: It should be H3 to be consistent with other parts of the readme

File: readme.md [hunk 6]
```diff
@@ -409,6 +414,49 @@ For more information, see the [Server API documentation](https://dev.fingerprint
...
+  {
+    key: Buffer.from('p2PA7MGy5tx56cnyJaFZMr96BCFwZeHjZV2EqMvTq53=', 'base64'),
+    algorithm: DecryptionAlgorithm.Aes256Gcm,
+  },
+]).then(result => {
+  console.log(result);
+});
+```
+
+### Params
```
Comment (ilfa) [inline readme.md:438]: It should be H4

File: src/sealedResults.ts [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+const SEALED_HEADER = Buffer.from([0x9e, 0x85, 0xdc, 0xed]);
+
+export function parseEventsResponse(unsealed: string) {
+  const json = JSON.parse(unsealed) as EventResponse;
+
+  if (!json.products) {
+    throw new Error('Sealed data is not valid events response');
+  }
```
Comment (ilfa) [inline src/sealedResults.ts:25]: We can not cast blindly, but try to use a [typeguard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types) here. WDYT?

**Review by ilfa — CHANGES_REQUESTED** [review]

File: src/sealedResults.ts (hunk 7 above)
Comment (TheUnderScorer) [reply src/sealedResults.ts]: Good idea, done!

File: readme.md (hunk 6 above)
Comment (TheUnderScorer) [reply readme.md]: Done!

File: readme.md (hunk 5 above)
Comment (TheUnderScorer) [reply readme.md:419]: Done!

Comment (ilfa) [reply readme.md:419]: Sorry for misunderstanding, but you made a wrong header a H3. [Here](https://github.com/fingerprintjs/fingerprintjs-pro-server-api-node-sdk/blob/main/readme.md?plain=1#L168) is what I mean

Comment (TheUnderScorer) [reply readme.md]: Ah, I see. It should be fixed now then 😁

**Review by ilfa — APPROVED** [review]
